contracts/vaults/RibbonDeltaVault.sol

```diff
@@ -122,6 +125,41 @@ contract RibbonDeltaVault is RibbonVault, OptionsDeltaVaultStorage {
         optionAllocationPct = _optionAllocationPct;
     }
 
+    /**
+     * @notice Updates the price per share of the current round. The current round
+     * pps will change right after call rollToNextOption as the gnosis auction contract
+     * takes custody of a % of `asset` tokens, and right after we claim the tokens from
+     * the action as we may recieve some of `asset` tokens back alongside the oToken,
+     * depending on the gnosis auction outcome. Finally it will change at the end of the week
+     * if the oTokens are ITM
+     */
+    modifier updatePPS(bool isWithdraw) {
+        if (!isWithdraw) {
+            _;
+        }
+
+        if (
+            !isWithdraw ||
+            roundPricePerShare[vaultState.round] <= PLACEHOLDER_UINT
```

Comment:
I included case where we are doing deposit/withdraw sequence for very first time in vault. we will have to update price per share in this case. but we will have to do this check for every withdraw afterwards (because rollToNextOption will be called)